src/map/clif.cpp

```diff
@@ -1072,6 +1072,72 @@ static int32 clif_setlevel(struct block_list* bl) {
 	return lv;
 }
 
+/*==========================================
+ * Gets guild ID for emblem display purposes only
+ *------------------------------------------*/
+static int32 clif_get_guild_id_for_emblem(struct block_list *bl) {
```

Comment:
Much better now. But it looks to me that whenever the guild_id is required for an emblem, the emblem_id is also required.

Why not make it a function that sets both using references?

Something like:
```cpp
static void clif_get_emblem_data(struct block_list *bl, int32& guild_id, int32& emblem_id) {
```
And then you can set it for example in the default case like this:
```cpp
guild_id = status_get_guild_id(bl);
emblem_id = status_get_emblem_id(bl);
```

Also please make sure to use the modern function comment style that explains all parameters and return value (if exists).
```cpp
/**
 * Description
 * @param bl [...]
 * @param guild_id [...]
 * @param emblem_id [...]
 */
```